Agent: add checks that DBus setup works properly

tobsan opened the pull request:
Makes sure that we can connect to the bus, and that we can
own the softwarecontainer name on that bus. In case of error,
the agent quits.

Also some cleanup in the generated code.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
Agent: add checks that DBus setup works properly

Makes sure that we can connect to the bus, and that we can
own the softwarecontainer name on that bus. In case of error,
the agent quits.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

breakreturn commented on agent/src/main.cpp:
```diff
@@ -216,11 +216,15 @@ int main(int argc, char **argv)
         ::softwarecontainer::SoftwareContainerAgent agent(mainContext, config, factory, utility);
-        std::unique_ptr<SoftwareContainerAgentAdaptor> adaptor(new SoftwareContainerAgentAdaptor(agent, useSessionBus));
+        std::unique_ptr<SoftwareContainerAgentAdaptor> adaptor(
+            new SoftwareContainerAgentAdaptor(ml, agent, useSessionBus)
```
Maybe there should be a comment what the agent adaptor needs the mainloop for? The reader can ofc look at the implementation but I'd like to see that already here if I read this.

breakreturn reviewed (approved):
Minor comment, do as you wish with that. Approving regardless.

thenor commented on agent/src/dbus/softwarecontaineragent_dbus_stub.h:
```diff
@@ -91,14 +92,23 @@ class SoftwareContainerAgent {
-private:
-bool emitSignal(const std::string& propName, Glib::VariantBase& value);
+    /*
+     * We emit this signal when we have a lost/acquired a name on dbus,
```
It's a list of three signals so "this" doesn't feel right to me. Just a nitpicky comment, but might as well do it right. ;)

kursatkobya reviewed (approved):
good addition with nice format corrections

thenor commented on agent/src/dbus/softwarecontaineragent_dbus_stub.cpp:
```diff
@@ -227,51 +242,51 @@ void com::pelagicore::SoftwareContainerAgent::on_method_call(const Glib::RefPtr<
-              Glib::ustring(),
-              Glib::Variant<std::vector<Glib::VariantBase> >::create_tuple(paramsList));
-      }
+    m_connection->emit_signal(
```
Is this a tab?

tobsan replied:
I think my editor converts tabs to spaces for me, plus I think everything was spaces already.

EDIT: That's a no :)

Commit:
servicetest: add session bus test

Adds a test that makes sure the agent fails if one tries to run on the
session bus without there being a bus available. This required
reworking the agent fixture to have class-scope, which in turn
required some timing checks to be added or moved.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>
